k9mail/src/main/java/com/fsck/k9/activity/compose/RecipientLoader.java

```diff
@@ -173,13 +174,62 @@ private static String getContactIdFromContactUri(Uri contactUri) {
     }
 
 
+    private Cursor getNicknameCursor(String nickname) {
+        Uri queryUriForNickname = ContactsContract.Data.CONTENT_URI;
+
+        final String[] PROJECTION_NICKNAME =
+                new String[] { ContactsContract.Data.CONTACT_ID, ContactsContract.CommonDataKinds.Nickname.NAME };
+
+        return getContext().getContentResolver().query(queryUriForNickname,
+                PROJECTION_NICKNAME,
+                ContactsContract.CommonDataKinds.Nickname.NAME + " LIKE ? AND " +
+                        Data.MIMETYPE + " = ?",
+                new String[] { nickname, ContactsContract.CommonDataKinds.Nickname.CONTENT_ITEM_TYPE },
+                null);
+    }
+
+    private void fillContactDataFromNickname(String nickname, List<Recipient> recipients,
+            Map<String, Recipient> recipientMap) {
+
+        Uri queryUri = ContactsContract.CommonDataKinds.Email.CONTENT_URI;
+        final ContentResolver contentResolver = getContext().getContentResolver();
+
+        final int INDEX_CONTACT_ID = 0;
+        final int INDEX_NICKNAME = 1;
+
+        Cursor nicknameCursor = getNicknameCursor(nickname);
+
+        if (nicknameCursor == null) {
+            return;
+        }
+
+        if (nicknameCursor.moveToFirst()) {
+            while (!nicknameCursor.isAfterLast()) {
+                String id = nicknameCursor.getString(INDEX_CONTACT_ID);
+                String selection = ContactsContract.Data.CONTACT_ID + " = ?";
+                Cursor cursor = contentResolver
+                        .query(queryUri, PROJECTION, selection, new String[] { id }, SORT_ORDER);
+
+                fillContactDataFromCursor(cursor, recipients, recipientMap, nicknameCursor.getString(INDEX_NICKNAME));
+
+                nicknameCursor.moveToNext();
+            }
+        }
+
+        nicknameCursor.close();
+    }
+
+
     private void fillContactDataFromQuery(String query, List<Recipient> recipients,
```

Comment:
This method confused me because it 1) delegates part of the filling to `fillContactDataFromNickname`, and 2) does part of the filling itself with a query. The second part should be extracted into its own method, which should leave `fillContactDataFromQuery` only calling to these two methods, which gives a clearer picture of what it does.